contactql/es/query.go

```diff
@@ -256,6 +256,35 @@ func attributeConditionToElastic(env envs.Environment, c *contactql.Condition) e
 		default:
 			panic(fmt.Sprintf("unsupported created_on attribute operator: %s", c.Operator()))
 		}
+	case contactql.AttributeLastSeenOn:
+		// special case for set/unset
+		if (c.Operator() == contactql.OpEqual || c.Operator() == contactql.OpNotEqual) && value == "" {
+			query = elastic.NewExistsQuery("last_seen_on")
+			if c.Operator() == contactql.OpEqual {
+				query = not(query)
+			}
+			return query
+		}
+
+		value := c.ValueAsDate()
+		start, end := dates.DayToUTCRange(value, value.Location())
+
+		switch c.Operator() {
+		case contactql.OpEqual:
+			return elastic.NewRangeQuery("last_seen_on").Gte(start).Lt(end)
+		case contactql.OpNotEqual:
+			return not(elastic.NewRangeQuery("last_seen_on").Gte(start).Lt(end))
```

Comment:
So `last_seen` can be null, is this going to behave the way we expect (or the user's expect) for these range queries? IE, does Elastic interpret `null` as a value? Notice that for the field case we check that that field is set as part of the query but that may be due to their structure. Could be worth trying out manually if you haven't already. (to be clear I *think* user expectation would be that last_seen comparisons always excluded null values)

Reply:
I checked that when I was updating rp-indexer - NULLs just aren't returned by range queries https://github.com/nyaruka/rp-indexer/blob/d85bf42559aa2afe71f91d39036b2e694fe2cd16/indexer_test.go#L102-L107

So this should behave the same as custom date field with no value